### ISSUE TITLE
[regression] Fastlane 2.74.0 breaks CircleCI archiving

### ISSUE THREAD

**Issue by jbelkins** [fastlane/fastlane#11511]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I have a fastlane task that archives an iOS app and generates an .ipa file from it.  It has run on CircleCI fine since it was created a few months ago.

Today I updated fastlane with no other changes to my project or its configuration, and my archive step started failing with no visible error message; xcodebuild exits with code 65 when archiving & exporting.  On further investigation I found that the change that specifically causes my failure is something between fastlane versions 2.73.0 and 2.74.0.  It also fails in the same manner when running the latest Fastlane version (2.75.1 at the moment.)

Archiving succeeds on my local machine when running the same fastlane version(s) & task that fails on CircleCI.

Here are the relevant actions from my Fastfile (name of app changed for posting):
```
    setup_circle_ci
    match(app_identifier: "com.bundle.identifier", type: "development", force_for_new_devices: true)
    gym(scheme: "AppNameAlpha", include_bitcode: false, export_method: "development", xcargs: xcargs, export_options: {
      uploadBitcode: false,
      uploadSymbols: true,
      compileBitcode: false })
```

Here is the generated xcodebuild command (again, name of app changed):
```
$ set -o pipefail && xcodebuild -workspace ./AppName.xcworkspace -scheme AppNameAlpha -destination 'generic/platform=iOS' -archivePath /Users/distiller/Library/Developer/Xcode/Archives/2018-01-09/AppName\ 2018-01-09\ 15.27.51.xcarchive APP_BUILD_NUMBER=1234 GIT_COMMIT_SHA=deadbeef HOCKEYAPP_APP_ID=0123456789ABCDEF BUNDLE_DISPLAY_NAME=AppName archive | tee /Users/distiller/Library/Logs/gym/AppName-AppNameAlpha.log | xcpretty
```

I do see some new messaging about checking your settings when you use export_method="development" but that does not seem to be the cause of the failure.

(Logs follow.  Preceding compilation steps omitted)
```
▸ Processing AppName-Info.plist
▸ Generating 'AppNameAlpha.app.dSYM'
▸ Running script 'Configure Google Services .plist'
▸ Running script 'Disable ATS for non-release builds'
▸ Running script '[CP] Embed Pods Frameworks'
▸ Running script '[CP] Copy Pods Resources'
▸ Touching AppNameAlpha.app
▸ Signing /Users/distiller/Library/Developer/Xcode/DerivedData/AppName-hdnmqadtpuogjabczdcachzltcjp/Build/Intermediates.noindex/ArchiveIntermediates/AppNameAlpha/InstallationBuildProductsLocation/Applications/AppNameAlpha.app
▸ Touching AppNameAlpha.app.dSYM
** ARCHIVE FAILED **
[15:32:59]: Exit status: 65
[15:32:59]: 
[15:32:59]: Maybe the error shown is caused by using the wrong version of Xcode
[15:32:59]: Found multiple versions of Xcode in '/Applications/'
[15:32:59]: Make sure you selected the right version for your project
[15:32:59]: This build process was executed using '/Applications/Xcode-9.2.app'
[15:32:59]: If you want to update your Xcode path, either
[15:32:59]: 
[15:32:59]: - Specify the Xcode version in your Fastfile
[15:32:59]: ▸ xcversion(version: "8.1") # Selects Xcode 8.1.0
[15:32:59]: 
[15:32:59]: - Specify an absolute path to your Xcode installation in your Fastfile
[15:32:59]: ▸ xcode_select "/Applications/Xcode8.app"
[15:32:59]: 
[15:32:59]: - Manually update the path using
[15:32:59]: ▸ sudo xcode-select -s /Applications/Xcode.app
[15:32:59]: 

+---------------+-----------------------------+
|              Build environment              |
+---------------+-----------------------------+
| xcode_path    | /Applications/Xcode-9.2.app |
| gym_version   | 2.74.0                      |
| export_method | development                 |
| sdk           | iPhoneOS11.2.sdk            |
+---------------+-----------------------------+

[15:32:59]: ▸ Touch /Users/distiller/Library/Developer/Xcode/DerivedData/AppName-hdnmqadtpuogjabczdcachzltcjp/Build/Intermediates.noindex/ArchiveIntermediates/AppNameAlpha/BuildProductsPath/Alpha-iphoneos/AppNameAlpha.app.dSYM
[15:32:59]: ▸     cd /Users/distiller/project
[15:32:59]: ▸     export PATH="/Applications/Xcode-9.2.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode-9.2.app/Contents/Developer/usr/bin:/Users/distiller/project/vendor/bundle/ruby/2.4.0/bin:/Users/distiller/.gem/ruby/2.4.2/bin:/Users/distiller/.rubies/ruby-2.4.2/lib/ruby/gems/2.4.0/bin:/Users/distiller/.rubies/ruby-2.4.2/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
[15:32:59]: ▸     /usr/bin/touch -c /Users/distiller/Library/Developer/Xcode/DerivedData/AppName-hdnmqadtpuogjabczdcachzltcjp/Build/Intermediates.noindex/ArchiveIntermediates/AppNameAlpha/BuildProductsPath/Alpha-iphoneos/AppNameAlpha.app.dSYM
[15:32:59]: 
[15:32:59]: ⬆️  Check out the few lines of raw `xcodebuild` output above for potential hints on how to solve this error
[15:32:59]: 📋  For the complete and more detailed error log, check the full log at:
[15:32:59]: 📋  /Users/distiller/Library/Logs/gym/AppName-AppNameAlpha.log
[15:32:59]: 
[15:32:59]: Your `export_method` in gym is defined as `development`
[15:32:59]: which might cause problems when signing your application
[15:32:59]: Are you sure want to build and export for development?
[15:32:59]: Please make sure to define the correct export methods when calling
[15:32:59]: gym in your Fastfile or from the command line
[15:32:59]: 
[15:32:59]: 
[15:32:59]: Looks like fastlane ran into a build/archive error with your project
[15:32:59]: It's hard to tell what's causing the error, so we wrote some guides on how
[15:32:59]: to troubleshoot build and signing issues: https://docs.fastlane.tools/codesigning/getting-started/
[15:32:59]: Before submitting an issue on GitHub, please follow the guide above and make
[15:32:59]: sure your project is set up correctly.
[15:32:59]: fastlane uses `xcodebuild` commands to generate your binary, you can see the
[15:32:59]: the full commands printed out in yellow in the above log.
[15:32:59]: Make sure to inspect the output above, as usually you'll find more error information there
```

A review of the raw logs shows no error messages that I can find

### Environment (run on CircleCI)

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                           |
| --------------------------- | ----------------------------------------------- |
| OS                          | 10.12.6                                         |
| Ruby                        | 2.4.2                                           |
| Bundler?                    | true                                            |
| Git                         | git version 2.14.3 (Apple Git-98)               |
| Installation Source         | ~/project/vendor/bundle/ruby/2.4.0/bin/fastlane |
| Host                        | Mac OS X 10.12.6 (16G29)                        |
| Ruby Lib Dir                | ~/.rubies/ruby-2.4.2/lib                        |
| OpenSSL Version             | OpenSSL 1.0.2m  2 Nov 2017                      |
| Is contained                | false                                           |
| Is homebrew                 | false                                           |
| Is installed via Fabric.app | false                                           |
| Xcode Path                  | /Applications/Xcode-9.2.app/Contents/Developer/ |
| Xcode Version               | 9.2                                             |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.74.0  | 🚫 Update available |

### Loaded fastlane plugins:

| Plugin                  | Version | Update-Status |
| ----------------------- | ------- | ------------- |
| fastlane-plugin-bugsnag | 1.1.0   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| bundler                   | 1.16.0       |
| io-console                | 0.4.6        |
| CFPropertyList            | 2.3.6        |
| concurrent-ruby           | 1.0.5        |
| i18n                      | 0.9.1        |
| minitest                  | 5.11.1       |
| thread_safe               | 0.3.6        |
| tzinfo                    | 1.2.4        |
| activesupport             | 4.2.10       |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| babosa                    | 1.0.2        |
| claide                    | 1.0.2        |
| fuzzy_match               | 2.0.4        |
| nap                       | 1.1.0        |
| cocoapods-core            | 1.3.1        |
| cocoapods-deintegrate     | 1.0.1        |
| cocoapods-downloader      | 1.1.3        |
| cocoapods-plugins         | 1.0.0        |
| cocoapods-search          | 1.0.0        |
| cocoapods-stats           | 1.0.0        |
| netrc                     | 0.11.0       |
| cocoapods-trunk           | 1.3.0        |
| cocoapods-try             | 1.1.0        |
| colored2                  | 3.1.2        |
| escape                    | 0.0.4        |
| fourflusher               | 2.0.1        |
| gh_inspector              | 1.0.3        |
| molinillo                 | 0.5.7        |
| ruby-macho                | 1.1.0        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.4        |
| cocoapods                 | 1.3.1        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.5        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| digest                    | 0.0.1        |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| dotenv                    | 2.2.1        |
| excon                     | 0.60.0       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.13.1       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.12.2       |
| fastimage                 | 2.1.1        |
| jwt                       | 2.1.0        |
| little-plugger            | 1.1.4        |
| multi_json                | 1.13.0       |
| logging                   | 2.2.2        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.8.1        |
| googleauth                | 0.6.2        |
| httpclient                | 2.8.3        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.1        |
| google-api-client         | 0.13.6       |
| json                      | 2.1.0        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.4.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| slack-notifier            | 1.5.1        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.3.0        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.6.4        |
| tty-cursor                | 0.5.0        |
| tty-spinner               | 0.7.0        |
| word_wrap                 | 1.0.0        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| xcpretty-travis-formatter | 1.0.0        |
| fastlane-plugin-bugsnag   | 1.1.0        |
| sqlite3                   | 1.3.13       |
| xcode-install             | 2.3.1        |
</details>

*generated on:* **2018-01-09**
</details>

Thanks for your help, let me know if you need more info

**Comment by KrauseFx** [comment]
Can you share
```
/Users/distiller/Library/Logs/gym/AppName-AppNameAlpha.log
```

**Comment by jbelkins** [comment]
@KrauseFx [AppName-AppNameAlpha.log.zip](https://github.com/fastlane/fastlane/files/1619547/AppName-AppNameAlpha.log.zip)

**Comment by jbelkins** [comment]
Discovered today that my problem was due to an interaction between a build script in my project and Fastlane source files, which CircleCI writes into the project directory during build setup at `vendor/`.  I modified my build script to ignore the `vendor/` directory and my build completed successfully.

Long story short, now I know to grep for this in my log file when xcodebuild exits with code 65 and no clear cause:
```
Command /bin/sh emitted errors but did not return a nonzero exit code to indicate failure
```

@KrauseFx Thanks for your assistance, no change to Fastlane is required.  Closing this ticket.
